### ISSUE TITLE
Federated statuses have "application" set to null

### ISSUE THREAD

**Issue by nolanlawson** [mastodon/mastodon#8476]
This seems to be a known issue, but I thought I'd open an issue to track it.

A status from instance A has the JSON data:

```json
{
  "application": {
    "name": "Foo App",
    "website": "https://foo.com"
  }
}
```

Users on instance A see this data. But when users on instance B see this status, it has `{"application": null}`. So they have to click on the permalink to see what the client application is.

Reproducible on v2.4.5.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by NicolasConstant** [comment]
Reproducible on v2.7.1.

**Comment by mjankowski** [comment]
This is pretty old and there's not a full set of URLs/context in the example ... but if this where I think it is, I think this is intentional and that the current behavior is to show that application if either the status author has set that value to true, or the user making the request is also the author.

If I've misunderstood, can you confirm still an issue on 4.3, and add some URLs/details?

**Comment by NicolasConstant** [comment]
Still reproducible on 4.4.0.

The issue is that the "Application" property isn't federated. 
For example, this status posted with the app Sengi : https://mastodon.social/@sengi_app/113552633850923148 

The application is visible in UI and API:
![Image](https://github.com/user-attachments/assets/e9aae229-e0e3-4805-8fba-878d982f43ad)
![Image](https://github.com/user-attachments/assets/7c94c457-8678-4543-b155-37c06de0f216)

But once federated and viewed from another instance, this property is lost:
![Image](https://github.com/user-attachments/assets/5b1e2650-af07-445b-9b05-2562e4539f22)
![Image](https://github.com/user-attachments/assets/7bdb4966-9b67-4bab-89bb-24a7ffc45c27)

The only way for the end user to get the info, is browsing manually to the original post/instance. Not great.

**Comment by trwnh** [comment]
There's been some talk of using something like `generator` to signal this:

```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "type": "Note",
  "content": "Posted from Foo App",
  "generator": {
    "name": "Foo App",
    "url": "https://foo.com"
  }
}
```

but the issue with this is that it's not technically correct; Foo App did not generate the ActivityStreams document. The more correct thing to do at a federation level would probably be to define an extension property for this, and reserve `generator` for some metadata representing the instance itself:

```json
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    {
      "mastodon_client": "http://joinmastodon.org/ns#application"
    }
  ],
  "type": "Note",
  "content": "Posted from Foo App",
  "mastodon_client": {
    "name": "Foo App",
    "url": "https://foo.com"
  },
  "generator": {
    "id": "https://mastodon.social/internal/service",
    "type": "Service",
    "name": "mastodon.social",
    "summary": "Mastodon v4.4.0-nightly.2024-11-26 running on mastodon.social",
    "url": "https://mastodon.social"
  }
}
```

**Comment by mjankowski** [comment]
> The more correct thing to do at a federation level would probably be to define an extension property for this, and reserve generator for some metadata representing the instance itself.

Makes sense to me, will leave this open as a satisfying explanation (if not solution) of the behavior in question, and as possible future ref if such an extension were to suddenly and miraculously emerge into the federalized omniverse.
